src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -1,715 +1,29 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
-
-using System;
-using System.Collections.Concurrent;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using System.Threading;
-using System.Threading.Tasks;
+using System.Web;
 using System.Web.Configuration;
-using NuGet.Server.Logging;
 
 namespace NuGet.Server.Infrastructure
 {
-    /// <summary>
-    /// ServerPackageRepository represents a folder of nupkgs on disk. All packages are cached during the first request in order
-    /// to correctly determine attributes such as IsAbsoluteLatestVersion. Adding, removing, or making changes to packages on disk 
-    /// will clear the cache.
-    /// </summary>
-    public class ServerPackageRepository
-        : PackageRepositoryBase, IServerPackageRepository, IPackageLookup, IDisposable
+	public class ServerPackageRepository : ServerPackageRepositoryCore
```

Comment:
It would be trivial to replace the getSetting Func<,,,> callback with an ISettingsProvider (with bool GetSetting(string settingName, bool defaultValue)) and it feels a little more "correct" (more like .net .NET, less like JavaScript.. :)) 

But a SettingsProvider that is limited to boolean values is probably not very futureproof. And since keys are dynamic it can be cumbersome for consumers to discover what settings are available.  

I wonder if something like this would be both simpler and cleaner:

`IServerRepositorySettings
{
  bool AllowOverrideExistingPackageOnPush {get;}
  bool IgnoreSymbolsPackages {get;}
  bool EnableDelisting {get;}
  bool EnableFrameworkFiltering {get;}
}

DefaultServerRepositorySettings : IServerRepositorySettings
{
  public virtual bool AllowOverrideExistingPackageOnPush  { get { return true;}}
  ...
  ...
  ...
}

WebServerRepositorySettings : : IServerRepositorySettings
{
  public virtual bool AllowOverrideExistingPackageOnPush  { get { return _cfgManager.AppSettings["AllowOverrideExistingPackageOnPush  "};}}
...
...
...
}
`
Or am I missing something? Maybe there was is a reason for the dynamic implementation?